# picard_markduplicates

## Description
Locate and tag duplicate reads in a BAM file

## Keywords
markduplicates, pcr, duplicates, bam, sam, cram

## Tool
- **picard**
  - description: A set of command line tools (in Java) for manipulating high-throughput sequencing (HTS)
data and formats such as SAM/BAM/CRAM and VCF.

  - homepage: https://broadinstitute.github.io/picard/
  - documentation: https://broadinstitute.github.io/picard/
  - licence: MIT

## Inputs
- **meta**
  - type: map
  - description: Groovy Map containing sample information
e.g. [ id:'test', single_end:false ]
- **reads**
  - type: file
  - pattern: *.{bam,cram,sam}
  - description: Sequence reads file, can be SAM/BAM/CRAM format
- **meta2**
  - type: map
  - description: Groovy Map containing reference information
e.g. [ id:'genome' ]
- **fasta**
  - type: file
  - pattern: *.{fasta,fa}
  - description: Reference genome fasta file, required for CRAM input
- **meta3**
  - type: map
  - description: Groovy Map containing reference information
e.g. [ id:'genome' ]
- **fai**
  - type: file
  - pattern: *.{fai}
  - description: Reference genome fasta index

## Outputs
- **meta**
  - type: map
  - description: Groovy Map containing sample information
e.g. [ id:'test', single_end:false ]
- **bam**
  - type: file
  - pattern: *.{bam}
  - description: BAM file with duplicate reads marked/removed
- **bai**
  - type: file
  - pattern: *.{bai}
  - description: An optional BAM index file. If desired, --CREATE_INDEX must be passed as a flag
- **cram**
  - type: file
  - pattern: *.{cram}
  - description: Output CRAM file
- **metrics**
  - type: file
  - pattern: *.{metrics.txt}
  - description: Duplicate metrics file generated by picard
- **versions**
  - type: file
  - pattern: versions.yml
  - description: File containing software versions

## Usage
This directory contains a Nextflow module.

- Module script: `main.nf`
- Metadata: `meta.yml`
- Tests: `tests/`

Import and call the module from a Nextflow workflow using your usual module include pattern.

## Task extensions
This module may support per invocation options via `task.ext.*`.

If you extend this module, document new `task.ext.*` keys here in the README.

## Provenance
- repository: https://github.com/broadinstitute/picard
- module_source: nf-core/modules
- original_authors: @drpatelh

This module was originally developed in the nf-core ecosystem and
has been adapted and versioned for the splice-nf-modules repository.

## Contacts
- authors: @drpatelh, @HR-cruk
- maintainers: @HR-cruk, @siddharthjayaraman

---

This README is generated from `meta.yml`. If you need to update it, edit `meta.yml` and rerun the generator.
